src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp

```diff
@@ -0,0 +1,409 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/PassTemporaryToStencilFunction.h"
+#include "dawn/Optimizer/AccessComputation.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/Optimizer/StatementMapper.h"
+#include "dawn/Optimizer/Stencil.h"
+#include "dawn/Optimizer/StencilInstantiation.h"
+#include "dawn/SIR/AST.h"
+#include "dawn/SIR/ASTVisitor.h"
+#include "dawn/SIR/SIR.h"
+
+namespace dawn {
+
+/// @brief this pass will identify temporaries of a stencil and replace their pre-computations
+/// by a stencil function. Each reference to the temporary is later replaced by the stencil function
+/// call.
+/// * Input: well formed SIR and IIR with the list of mss/stages, temporaries used and
+/// <statement,accesses> pairs
+/// * Output: modified SIR, new stencil functions are inserted and calls. Temporary fields are
+/// removed. New stencil functions instantiations are inserted into the IIR. <statement,accesses>
+/// pairs are recomputed
+
+namespace {
+sir::Interval intervalToSIRInterval(Interval interval) {
+  return sir::Interval(interval.lowerLevel(), interval.upperLevel(), interval.lowerOffset(),
+                       interval.upperOffset());
+}
+
+/// @brief some properties of the temporary being replaced
+struct TemporaryFunctionProperties {
+  std::shared_ptr<StencilFunCallExpr>
+      stencilFunCallExpr_;        /// stencil function call that will replace the tmp
```

Comment:
ok, what is the meaning, use of `< `?

Reply:
from https://www.stack.nl/~dimitri/doxygen/manual/docblocks.html

If you want to document the members of a file, struct, union, class, or enum, it is sometimes desired to place the documentation block after the member instead of before. For this purpose you have to put an additional < marker in the comment block. Note that this also works for the parameters of a function. [...]

`int var; ///< Brief description after the member`